src/plugins/intel_gpu/src/graph/graph_optimizer/fuse_primitives.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "program_helpers.h"
+#include "pass_manager.h"
+
+#include "eltwise_inst.h"
+#include "quantize_inst.h"
+
+#include <vector>
+#ifdef ENABLE_ONEDNN_FOR_GPU
+#include <impls/onednn/utils.hpp>
+#endif
+
+using namespace cldnn;
+
+static bool eltwise_supports_fusings(eltwise_node& node) {
+    auto out_layout = node.get_output_layout();
+    if (out_layout.data_type == data_types::f16 && out_layout.batch() > 1 && out_layout.format == format::fs_b_yx_fsv32) {
+        return false;
+    }
+
+    return true;
+}
+
+void fuse_primitives::run(program& p) {
+    bool need_recalc_processing_order = false;
+    std::map<primitive_id, std::vector<std::pair<primitive_id, size_t>>> fusing_history;
+
+        auto itr = p.get_processing_order().begin();
+    while (itr != p.get_processing_order().end()) {
+        auto node_itr = itr++;
+        auto& node = (*node_itr);
+
+        if (node->is_output() || node->is_constant())
+            continue;
+
+        auto func_fuse_quantize = [&](quantize_node& node) {
+            bool should_fuse = false;
+            auto out_layout = node.get_output_layout();
+            if (out_layout.is_dynamic() || node.is_in_shape_of_subgraph())
+                return;
+
+            auto& input_node = node.get_dependency(0);
+            auto in_layout = input_node.get_output_layout();
+            if (in_layout.is_dynamic() || input_node.is_in_shape_of_subgraph())
+                return;
+
+            should_fuse |= input_node.is_type<eltwise>() && eltwise_supports_fusings(input_node.as<eltwise>());
+            should_fuse |= node.get_scale_shift_opt();
```

Comment:
this line seems weird. If `node` has `scale_shift_opt`, it can be just merged?

Reply:
That's right. removed.